Geometry_Engine/Query/Normal.cs

```diff
@@ -85,6 +85,111 @@ public static List<Vector> Normals(this Mesh mesh)
             return normals.ToList();
         }
 
+        /***************************************************/
+        /**** Public Methods - Curves                   ****/
+        /***************************************************/
+
+
+        public static Vector Normal(this Polyline curve)
+        {
+
+            if (!curve.IsPlanar())
+            {
+                Reflection.Compute.RecordError("Input must be planar.");
+                return null;
+            }
+            else if (!curve.IsClosed())
+            {
+                Reflection.Compute.RecordError("Curve is not closed. Input must be a polygon");
+                return null;
+            }
+            else if (curve.IsSelfIntersecting())
+            {
+                Reflection.Compute.RecordWarning("Curve is self intersecting");
+                return null;
+            }
+
+            Vector normal = new Vector { X = 0, Y = 0, Z = 0 };
+
+            for (int i = 0; i < curve.ControlPoints.Count - 1; i++)
+            {
+                Point pA = curve.ControlPoints[i];
+                Point pB = curve.ControlPoints[i + 1];
+                Point pC = curve.ControlPoints[(i + 2) % curve.ControlPoints.Count];
+
+                normal += CrossProduct(pB - pA, pC - pB);
+            }
+
+            return normal.Normalise();
+        }
+
+        /***************************************************/
+
+        public static Vector Normal(this PolyCurve curve)
+        {
+
+            if (!curve.IsPlanar())
+            {
+                Reflection.Compute.RecordError("Input must be planar.");
+                return null;
+            }
+            else if (!curve.IsClosed())
+            {
+                Reflection.Compute.RecordError("Curve is not closed. Input must be a polygon");
+                return null;
+            }
+            else if (curve.IsSelfIntersecting())
+            {
+                Reflection.Compute.RecordWarning("Curve is self intersecting");
+                return null;
+            }
+
+
+            Vector normal = new Vector { X = 0, Y = 0, Z = 0 };
+
+            List<Point> pts = new List<Point> { curve.Curves[0].IStartPoint() };
+
+            foreach (ICurve crv in curve.Curves)
```

Comment:
This part need to call curve.ISubParts() intead of just the proeprty of the curve if we are to assume only arcs and lines below. See testfile.